scripts/timelapse-colorizer-data/convert_emt_nuclear_data.py

```diff
@@ -0,0 +1,267 @@
+```
+python timelapse-colorizer-data/convert_emt_nuclear_data.py --scale 1.0 --output_dir=/allen/aics/animated-cell/Dan/fileserver/colorizer/EMT_nuclear
+```
+"""
+
+from typing import List
+from aicsimageio import AICSImage
+import argparse
+import json
+import logging
+import numpy as np
+import pandas as pd
+from pandas.core.groupby.generic import DataFrameGroupBy
+import time
+
+from data_writer_utils import (
+    INITIAL_INDEX_COLUMN,
+    RESERVED_INDICES,
+    ColorizerDatasetWriter,
+    FeatureMetadata,
+    configureLogging,
+    extract_units_from_feature_name,
+    scale_image,
+    remap_segmented_image,
+)
+
+# DATASET SPEC: See DATA_FORMAT.md for more details on the dataset format!
+# You can find the most updated version on GitHub here:
+# https://github.com/allen-cell-animated/nucmorph-colorizer/blob/main/documentation/DATA_FORMAT.md
+
+# OVERWRITE THESE!! These values should change based on your dataset. These are
+# relabeled as constants here for clarity/intent of the column name.
+OBJECT_ID_COLUMN = "Label"
+"""Column of object IDs (or unique row number)."""
+# TRACK_ID_COLUMN = ""
+"""Column of track ID for each object."""
+TIMES_COLUMN = "Frame"
+"""Column of frame number that the object ID appears in."""
+SEGMENTED_IMAGE_COLUMN = "Filepath"
+"""Column of path to the segmented image data or z stack for the frame."""
+CENTROIDS_X_COLUMN = "x"
+"""Column of X centroid coordinates, in pixels of original image data."""
+CENTROIDS_Y_COLUMN = "y"
+"""Column of Y centroid coordinates, in pixels of original image data."""
+FEATURE_COLUMNS = [
+    "Slice",
+    "Area",
+    "Orientation",
+    "Aspect_Ratio",
+    "Circularity",
+    "Mean_Fluor",
+]
+"""Columns of feature data to include in the dataset. Each column will be its own feature file."""
+
+FEATURE_COLUMNS_TO_UNITS = {
+    "Mean_Fluor": "AU",  # arbitrary units
+    "Area": "px²",
+}
+FEATURE_COLUMNS_TO_NAMES = {
+    "Aspect_Ratio": "Aspect Ratio",
+    "Mean_Fluor": "Mean Fluorescence",
+}
+
+
+def make_frames(
+    grouped_frames: DataFrameGroupBy,
+    scale: float,
+    writer: ColorizerDatasetWriter,
+):
+    """
+    Generate the images and bounding boxes for each time step in the dataset.
+    """
+    nframes = len(grouped_frames)
+    logging.info("Making {} frames...".format(nframes))
+
+    is_nonzero = lambda n: n != 0
+
+    for group_name, frame in grouped_frames:
+        start_time = time.time()
+
+        # Get the path to the segmented zstack image frame from the first row (should be the same for
+        # all rows in this group, since they are all on the same frame).
+        row = frame.iloc[0]
+        frame_number = row[TIMES_COLUMN]
+        # Flatten the z-stack to a 2D image.
+        zstackpath = row[SEGMENTED_IMAGE_COLUMN]
+        zstackpath = zstackpath.strip('"')
+        zstack = AICSImage(zstackpath).get_image_data("ZYX", S=0, T=0, C=0)
+        # min but ignore zero values. Doing `min(0, id)` will always return 0 which results
```

Comment:
I think one additional comment to say here would be: we are taking min projection because the objects were segmented such that lower indices are lower in Z, and the nature of the data is such that we want to those lower objects to have precedence when there are overlaps due to being projected into 2D.